extensions/lifecycle/conductor/LifecycleConductor.js

```diff
@@ -0,0 +1,270 @@
+                (bucket, done) => {
+                    const [ownerId, bucketName] = bucket.split(':');
+                    if (!ownerId || !bucketName) {
+                        this.logger.error(
+                            'malformed zookeeper bucket entry, skipping',
+                            { zkPath: zkBucketsPath, bucket });
+                        return process.nextTick(done);
+                    }
+                    return this._processBucket(ownerId, bucketName, done);
+                }, next),
+            (entries, next) => {
+                this.logger.debug(
+                    'producing kafka entries',
+                    { topic: this.lcConfig.bucketTasksTopic,
+                      entries });
+                if (entries.length === 0) {
+                    return next();
+                }
+                return this._producer.send(entries, next);
+            },
+        ], () => {
+            if (lockHeld) {
+                this._zkClient.remove(zkLockPath, -1, err => {
+                    if (err) {
+                        this.logger.error(
+                            'error removing conductor lock on zookeeper',
+                            { zkPath: zkLockPath, error: err.message });
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * Initialize kafka producer and zookeeper client
+     *
+     * @param {function} done - callback
+     * @return {undefined}
+     */
+    init(done) {
+        if (this._zkClient) {
+            // already initialized
+            return process.nextTick(done);
+        }
+        return async.series([
+            next => {
+                const producer = new BackbeatProducer({
+                    zookeeper: { connectionString:
+                                 this.zkConfig.connectionString },
+                    topic: this.lcConfig.bucketTasksTopic,
+                });
+                producer.once('error', next);
+                producer.once('ready', () => {
+                    producer.removeAllListeners('error');
+                    producer.on('error', err => {
+                        this.logger.error('error from backbeat producer', {
+                            topic: this.lcConfig.bucketTasksTopic,
+                            error: err,
+                        });
+                    });
+                    this._producer = producer;
+                    next();
+                });
+            },
+            next => {
+                this._zkClient = zookeeperHelper.createClient(
+                    this.zkConfig.connectionString);
+                this._zkClient.connect();
+                this._zkClient.once('error', next);
+                this._zkClient.once('ready', () => {
+                    // just in case there would be more 'error' events
+                    // emitted
+                    this._zkClient.removeAllListeners('error');
```

Comment:
is it not worth logging zookeeper errors like you do for the producer above? with like `this._zkClient.on('error', etc)`